cli/src/pcluster/api/controllers/cluster_operations_controller.py

```diff
@@ -305,16 +315,81 @@ def update_cluster(
-    return UpdateClusterResponseContent(
-        cluster=ClusterInfoSummary(
-            cluster_name="nameeee",
-            cloudformation_stack_status=CloudFormationStatus.CREATE_COMPLETE,
-            cloudformation_stack_arn="arn",
-            region="region",
-            version="3.0.0",
-            cluster_status=ClusterStatus.CREATE_COMPLETE,
+
+    # Validate inputs
+    if client_token:
+        raise BadRequestException("clientToken is currently not supported for this operation")
+    cluster_config = read_config(update_cluster_request_content.cluster_configuration)
+
+    try:
+        cluster = Cluster(cluster_name)
+        if not check_cluster_version(cluster, exact_match=True):
+            raise BadRequestException(
+                f"the update can be performed only with the same ParallelCluster version ({cluster.stack.version}) "
+                "used to create the cluster."
+            )
+
+        if dryrun:
+            cluster.validate_update_request(
+                target_source_config=cluster_config,
+                force=force_update,
+                validator_suppressors=get_validator_suppressors(suppress_validators),
+                validation_failure_level=FailureLevel[validation_failure_level],
+            )
+            raise DryrunOperationException()
+
+        changes, ignored_validation_failures = cluster.update(
+            target_source_config=cluster_config,
+            validator_suppressors=get_validator_suppressors(suppress_validators),
+            validation_failure_level=FailureLevel[validation_failure_level],
+            force=force_update,
         )
+
+        return UpdateClusterResponseContent(
+            cluster=ClusterInfoSummary(
+                cluster_name=cluster_name,
+                cloudformation_stack_status=CloudFormationStatus.UPDATE_IN_PROGRESS,
+                cloudformation_stack_arn=cluster.stack.id,
+                region=os.environ.get("AWS_DEFAULT_REGION"),
+                version=cluster.stack.version,
+                cluster_status=cloud_formation_status_to_cluster_status(CloudFormationStatus.UPDATE_IN_PROGRESS),
+            ),
+            validation_messages=validation_results_to_config_validation_errors(ignored_validation_failures) or None,
+            change_set=_get_changeset(changes),
+        )
+    except ConfigValidationError as e:
+        raise _handle_config_validation_error(e)
+    except ClusterUpdateError as e:
+        raise _handle_cluster_update_error(e)
+    except (NotExistsClusterActionError, StackNotFoundError):
+        raise NotFoundException(
+            f"cluster '{cluster_name}' does not exist or belongs to an incompatible ParallelCluster major version."
+        )
+
+
+def _get_changeset(changes):
+    if changes is None or len(changes) <= 1:
+        return None
+
+    key_indexes = {key: index for index, key in enumerate(changes[0])}
+    return [
+        Change(
+            parameter=row[key_indexes["parameter"]],
```

Comment:
let's show the full path for the parameter

Reply:
As agreed, we will compose it following the [YAML path specification](https://github.com/wwkimball/yamlpath/wiki/Segments-of-a-YAML-Path).